mysql-test/suite/rpl_gtid/t/rpl_wait_for_semi_sync_ack.test

```diff
@@ -0,0 +1,289 @@
+--source include/sync_slave_sql_with_master.inc
+
+--let $assert_text= Semi-sync Slave: should have t1, t2 and t3
+--let $assert_cond= [select count(*) from information_schema.tables where table_schema="test" and table_name in ("t1", "t2", "t3", "t4")] = 3
+--source include/assert.inc
+
+
+--let $rpl_connection_name= semi_sync_slave
+--source include/rpl_connection.inc
+set @@global.debug= '-d,before_semi_sync_reply';
+
+
+
+# Case 4: Lagging async slave + master restart
+
+--echo "Stopping async slave to simulate lag"
+--let $rpl_connection_name= async_slave
+--source include/rpl_connection.inc
+--source include/stop_slave.inc
+
+--echo "Generating traffic on the master"
+--source include/rpl_connection_master.inc
+create table t4(a int);
+insert into t4 values(1);
+insert into t4 values(2);
+flush logs;
+insert into t4 values(3);
+insert into t4 values(4);
+flush logs;
+
+--let $sync_slave_connection= semi_sync_slave
+--source include/sync_slave_sql_with_master.inc
+
+--source include/rpl_connection_master.inc
+--echo "Restarting master"
+--let $rpl_server_number=1
+--source include/rpl_restart_server.inc
+
+--echo "Starting async slave"
+--let $rpl_connection_name= async_slave
+--source include/rpl_connection.inc
+--source include/start_slave.inc
+
+--echo "Waiting till async slave is caught up"
+--source include/rpl_connection_master.inc
+--let $sync_slave_connection= async_slave
+--source include/sync_slave_sql_with_master.inc
+
+--let $assert_text= Async Slave: t4 should have 4 entries
+--let $assert_cond= [select count(*) from t4] = 4
+--source include/assert.inc
+
+
+
+# Case 5: Should wait for ACK after RESET MASTER
+
+--let $rpl_connection_name= semi_sync_slave
+--source include/rpl_connection.inc
+--source include/stop_slave.inc
+
+--let $rpl_connection_name= async_slave
+--source include/rpl_connection.inc
+--source include/stop_slave.inc
+
+--source include/rpl_connection_master.inc
+set @gtid_exec= @@global.gtid_executed;
+reset master;
+
+--let $last_acked= query_get_value(show status like "Rpl_last_semi_sync_acked_pos", Value, 1)
+--let $assert_text= Last acked pos should be empty
+--let $assert_cond= "$last_acked" = "./master-bin.000001:0"
+--source include/assert.inc
+
+set @@global.gtid_purged= @gtid_exec;
+--let $binlog= query_get_value(SHOW MASTER STATUS, File, 1)
+--replace_result $binlog binlog
+--eval purge binary logs to '$binlog'
+
+--let $rpl_connection_name= semi_sync_slave
+--source include/rpl_connection.inc
+--source include/start_slave.inc
+
+--let $rpl_connection_name= async_slave
+--source include/rpl_connection.inc
+--source include/start_slave.inc
+
+--let $rpl_connection_name= semi_sync_slave
+--source include/rpl_connection.inc
+set @@global.debug= '+d,before_semi_sync_reply';
+
+--source include/rpl_connection_master.inc
+--send create table t5 (a int)
+
+--let $rpl_connection_name= semi_sync_slave
+--source include/rpl_connection.inc
+set debug_sync='now WAIT_FOR reached';
+
+--sleep 1
```

Comment:
Doesn't look reliable. May be something like `sync_slave_io_with_master.inc` instead?